lib/active_model/serializer/adapter.rb

```diff
@@ -1,7 +1,10 @@
+require 'active_model/serializer/cache'
+
 module ActiveModel
   class Serializer
     class Adapter
       extend ActiveSupport::Autoload
+      include ActiveModel::Serializer::Cache
```

Comment:
Sounds good, that's a good start. I don't think, though, that another class will result in any significant loss.

***

On another not, I think we need to name things better here; we need to better group the code. For instance, I have no idea from a glance what the following is:

```ruby
klass._cache_only && !klass._cache_except || !klass._cache_only && klass._cache_except
```

I'd rather have


```ruby
def cached_object
  # ...
  elsif fragment_cache?
      FragmentCache.new(self, serializer, @options, @root).fetch
  else
  #...
end

private

def fragment_cache?
  klass._cache_only && !klass._cache_except || !klass._cache_only && klass._cache_except
end
```